main/main.c

```diff
@@ -611,6 +611,7 @@ PHP_INI_BEGIN()
 #ifdef PHP_WIN32
 	STD_PHP_INI_BOOLEAN("windows.show_crt_warning",		"0",		PHP_INI_ALL,		OnUpdateBool,			windows_show_crt_warning,			php_core_globals,	core_globals)
 #endif
+	STD_PHP_INI_ENTRY("syslog.highbit",			"0",		 PHP_INI_SYSTEM,	OnUpdateBool,		syslog_highbit,		php_core_globals, 		core_globals)
```

Comment:
> In fact, given the wording of the spec it would even be valid to come up with your very own encoding and use that, in which case PHP has absolutely no information about what are newlines and what are control characters. Going strictly by the spec and allowing everything it allows it would be illegal to perform any kind of processing on our side.

@nikic Well, almost.  RFC-5424 relaxes the constraints of RFC-3164 somewhat:

```
   [...] These values are legal, but a syslog application MAY modify these
   characters upon reception.  For example, it might change them into an
   escape sequence (e.g., value 0 may be changed to "\0").  A syslog
   application SHOULD NOT modify any other octet values.
```

And that's exactly what I do.  I write `#ooo` (3 octal digits) into the log.  I don't bother escaping the pound character itself, though I suppose I could to avoid ambiguity.